x-pack/plugins/task_manager/server/task_store.ts

```diff
@@ -252,14 +257,14 @@ export class TaskStore {
       return attrsById;
     }, new Map());
 
-    let updatedSavedObjects: Array<SavedObjectsUpdateResponse | Error>;
+    let updatedSavedObjects: Array<SavedObjectsUpdateResponse<SerializedConcreteTaskInstance>>;
     try {
       ({ saved_objects: updatedSavedObjects } =
         await this.savedObjectsRepository.bulkUpdate<SerializedConcreteTaskInstance>(
           docs.map((doc) => ({
             type: 'task',
             id: doc.id,
-            options: { version: doc.version },
+            version: doc.version,
```

Comment:
This is the bug fix I had to do to pass-trhough the doc's version number to the SO client to trigger 409 conflict errors. I don't understand how there isn't type safety here but worth investigating separately.